silnlp/nmt/config.py

```diff
@@ -642,6 +645,7 @@ def has_parent(self) -> bool:
     def set_seed(self) -> None:
         seed = self.data["seed"]
         set_seed(seed)
+        sp.set_random_generator_seed(seed)
```

Comment:
This won't give the expected behavior. You can only set it once for a process and it must be set before you use sentencepiece anywhere else (see sentencepiece [issue](https://github.com/google/sentencepiece/issues/635)). I would just set this for the tests.

Reply:
Fixed - called for each preprocessing test.